File: spring-web/src/main/java/com/example/springwebex/service/ActorService.java
package com.example.springwebex.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.example.springwebex.dao.DvdRentalDao;
import com.example.springwebex.model.rdb.ActorDto;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class ActorService {
    private final DvdRentalDao dao;

    public List<ActorDto> getActors() {
        return dao.selectActors();
    }

    public ActorDto getActorById(Long id) {
        return dao.selectActorById(id);
    }

    public void setActor(String firstName, String lastName) {
        var actor = new ActorDto();
        actor.setFirstName(firstName);
        actor.setLastName(lastName);
        dao.insertActor(actor);
    }
}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/service/ActorService.kt
package com.example.springwebex.service

import org.springframework.stereotype.Service
import com.example.springwebex.dao.DvdRentalDao
import com.example.springwebex.model.rdb.ActorDto

@Service
class ActorService(private val dao: DvdRentalDao) {

    fun getActors(): List<ActorDto> {
        return dao.selectActorsByPage(pageSize = 100, offset = 0)
    }

    fun getActorById(id: Long): ActorDto {
        return dao.selectActorById(id.toInt()) ?: ActorDto()
    }

    fun setActor(firstName: String, lastName: String) {
        val actor = ActorDto(firstName = firstName, lastName = lastName)
    }
}
